--- kernel/nvidia-uvm/uvm_linux.h	2016-07-12 08:52:17.000000000 +0400
+++ kernel/nvidia-uvm/uvm_linux_.h	2016-07-29 15:36:32.905204440 +0400
@@ -554,7 +554,7 @@
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
-static bool radix_tree_empty(struct radix_tree_root *tree)
+static bool radix_tree_is_empty(struct radix_tree_root *tree)
 {
     void *dummy;
     return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
